src/components/core/a11y/focus.ts

```diff
@@ -99,6 +111,32 @@ export class FocusTrap {
     );
   }
 
+  public setDataHasVisibleFocusWithin(element: HTMLElement): void {
```

Comment:
not sure if we on the method name should indicate that this will be constantly tracked, not only once?

Reply:
I'm not sure either, do you have any suggestion? `setDataHasVisibleFocusWithinListener`?